packages/blocks/src/components/portal.ts

```diff
@@ -150,6 +156,22 @@ export function createSimplePortal({
 
   updatePortal(updateId);
   container.append(portalRoot);
+
+  function _onClickAway(e: Event) {
+    if (portalRoot.contains(e.target as Node)) return;
+    dispose();
+  }
+
+  if (closeOnClickAway) {
+    // Avoid triggering click away listener on initial render
+    setTimeout(() => document.addEventListener('click', _onClickAway));
+  }
```

Comment:
Try using the following pattern; you no longer need to use `dispose`.

```ts
  function _onClickAway(e: Event) {
    if (portalRoot.contains(e.target as Node)) return;
    abortController.abort();
  }

document.addEventListener('click', _onClickAway, { signal: abortController.signal }));
```

By the way, I believe placing `closeOnClickAway` in `createLitPortal` may be more appropriate since I want to keep `createSimplePortal` as tidy as possible.